Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/organizer/create/TraceLocationCreateFragment.kt

```diff
@@ -0,0 +1,126 @@
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import org.joda.time.DateTimeZone
+import org.joda.time.LocalDate
+import org.joda.time.LocalDateTime
+import org.joda.time.LocalTime
+import javax.inject.Inject
+
+class TraceLocationCreateFragment : Fragment(R.layout.trace_location_create_fragment), AutoInject {
+
+    private val binding: TraceLocationCreateFragmentBinding by viewBindingLazy()
+    private val navArgs by navArgs<TraceLocationCreateFragmentArgs>()
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: TraceLocationCreateViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as TraceLocationCreateViewModel.Factory
+            factory.create(navArgs.category)
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.toolbar.setNavigationOnClickListener { popBackStack() }
+
+        viewModel.uiState.observe(viewLifecycleOwner) { state ->
+            binding.toolbar.setSubtitle(state.title)
+            binding.valueStart.text = state.getStartDate(requireContext().getLocale())
+            binding.valueEnd.text = state.getEndDate(requireContext().getLocale())
+            binding.layoutBeginn.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
+            binding.layoutEnd.visibility = if (state.isDateVisible) View.VISIBLE else View.GONE
+            binding.valueLengthOfStay.text = state.getLength(resources)
+            binding.buttonSubmit.isEnabled = state.isSendEnable
+        }
+
+        binding.descriptionInputEdit.doOnTextChanged { text, _, _, _ ->
+            viewModel.description = text?.toString()
+        }
+
+        binding.placeInputEdit.doOnTextChanged { text, _, _, _ ->
+            viewModel.address = text?.toString()
+        }
+
+        binding.layoutBeginn.setOnClickListener {
+            it.hideKeyboard()
+            showDatePicker(viewModel.start) { value ->
+                viewModel.start = value
+            }
+        }
+
+        binding.layoutEnd.setOnClickListener {
+            it.hideKeyboard()
+            showDatePicker(viewModel.end) { value ->
+                viewModel.end = value
+            }
+        }
+
+        binding.layoutLengthOfStay.setOnClickListener {
+            it.hideKeyboard()
+            showDurationPicker()
+        }
+
+        binding.buttonSubmit.setOnClickListener {
+            viewModel.send()
+        }
+    }
+
+    private fun showDatePicker(defaultValue: LocalDateTime?, callback: (LocalDateTime) -> Unit) {
+        MaterialDatePicker.Builder.datePicker().setSelection(defaultValue?.toDateTime(DateTimeZone.UTC)?.millis).build()
```

Comment:
Isn't the blue in the date picker too light? Should we use the blue from our `accent` color here?

Reply:
Updated, but I guess we should update styles to have one primary/accent color for all widgets.